contracts/reputation/Coreputation.sol

```diff
@@ -74,6 +74,47 @@ contract Coreputation is MasterCopyNonUpgradable, CoConsensusModule {
         CoConsensusModule.setupCoConsensus(_coConsensus);
     }
 
+    /**
+     * @notice Insert or update validator information.
+     *
+     * @dev Validator status is marked Staked, if validator status is
+     *      Undefined and he has non zero reputation value.
+     *      Validator is not inserted if his reputation value is zero.
+     *      Validator status is marked to Deregistered when his current status
+     *      is Staked and reputation value changes to zero.
+     *
+     * @dev Function requires:
+     *      - msg.sender should be CoConsensus
+     *      - Validator address should non zero
+     *      - Newly added validators reputation should be non zero
```

Comment:
this contradicts always succeeding an `upsert`; rather `Undefined->Deregisered` is allowed in `Coreputation::upsert()`

Reply:
Removed it. it was stale statement.